exonum-node/src/consensus.rs

```diff
@@ -537,7 +541,12 @@ impl NodeHandler {
                     .prevotes(prevote_round, propose_hash)
                     .iter()
                     .map(|p| p.clone().into());
-                self.blockchain.save_messages(round, raw_messages);
+
+                let fork = self.blockchain.fork();
```

Comment:
Pattern "fork -> do one thing -> merge" occurs several time in the code. Could it make sense to create a HOF for that, e.g.
```rust
fn persist_changes<F: FnOnce(&Fork)>(&self, changes: F) {
    let fork = self.blockchain.fork();
    changes(&fork);
    self.blockchain.merge(fork.into_patch()).expect("..");
}
```

?